packages/net/http/payload.pony

```diff
@@ -1,40 +1,102 @@
+Appication buffer sizes determine how much data is fed to the TCP connection
+at once, but the total amount must match this size.
+
+3. **ChunkedTransfer**.  This is used when the payload length can not be known
+in advance, but can be large.   It is selected by calling `Payload.set_length`
+with a parameter of `None`.  On the TCP link this mode can be detected because
+there is no `Content-Length` header at all, being replaced by the
+`Transfer-Encoding: chunked` header.  In addition, the message body is
+separated into chunks, each with its own bytecount.  As with `StreamTransfer`
+mode, transmission can be spread out over time with the difference that it is
+the original data source that determines the chunk size.
+
+If `Payload.set_length` is never called at all, a variation on
+`StreamTransfer` called `OneshotTransfer` is used.  In this case, all of the
+message body is placed into the message at once, using `Payload.add_chunk`
+calls.  The size will be determined when the message is submitted for
+transmission.  Care must be taken not to consume too much memory, especially
+on a server where there can be multiple messages in transit at once.
+
+The type of transfer being used by an incoming message can be determined
+from its `transfer_mode` field, which will be one of the
+[TransferMode](net-http-TransferMode) types.
+
+### Sequence
+
+For example, to send a message of possibly large size:
+
+1. Create the message with a call to `Payload.request` or `Payload.response`.
+2. Set the `session` field of the message.
+2. Call `Payload.set_length` to indicate the length of the body.
+3. Add any additional headers that may be required, such as `Content-type`.
+4. Submit the message for transmission by calling the either the
+`HTTPSession.apply` method (in servers) or the `HTTPCLient.apply` method
+in clients.
+5. Wait for the `send_body` notification.
+6. Make any number of calls to `Payload.send_chunk`.
+7. Call `Payload.finish`.
+
+To send a message of small, reasonable size (say, under 20KB), this
+simplified method can be used instead:
+
+1. Create the message with a call to `Payload.request` or `Payload.response`.
+2. Set the `session` field of the message.
+3. Add any additional headers that may be required, such as `Content-type`.
+4. Call `add_chunk` one or more times to add body data.
+4. Submit the message for transmission by calling the either the
+[HTTPSession](net-http-HTTPSession)`.apply` method (in servers) or the
+[HTTPClient](net-http-HTTPClient)`.apply` method in clients.
   """
-  var handler: (ResponseHandler | None)
   var proto: String = "HTTP/1.1"
   var status: U16
   var method: String
   var url: URL
-
+  var _body_length: USize = 0
+  var transfer_mode: TransferMode = OneshotTransfer
+  var session: (HTTPSession tag | None) = None
+  
   let _headers: Map[String, String] = _headers.create()
-  let _body: Array[ByteSeq] = _body.create()
+  var _body: Array[ByteSeq val] = _body.create()
```

Comment:
var field is never reassigned and is only initialized with a constructor, so it should be declared with embed.